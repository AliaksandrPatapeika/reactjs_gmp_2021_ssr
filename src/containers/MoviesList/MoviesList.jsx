import './MoviesList.less';

import React from 'react';
import {useSelector} from 'react-redux';

import MovieCard from '../MovieCard';

const MoviesList = () => {
  const movies = useSelector((state) => state.movie.movies);

  return (
    <div className="moviesListContainer">
      {movies.map((item) => (
        <MovieCard
          movie={item}
          key={item.id}
        />
      ))}
    </div>
  );
};

export default React.memo(MoviesList);
