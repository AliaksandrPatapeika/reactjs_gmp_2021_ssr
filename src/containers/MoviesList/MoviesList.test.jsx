import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import MockMovies from '../../tests/mocks/mockData';
import MoviesList from './MoviesList';

describe('MoviesList', () => {
  test('renders correctly', () => {
    const mockStore = configureMockStore();
    const initialState = {
      movie: {
        movies: MockMovies
      }
    };
    const store = mockStore(initialState);
    const moviesList = renderer
      .create(
        <Provider store={store}>
          <MoviesList />
        </Provider>
      )
      .toJSON();

    expect(moviesList)
      .toMatchSnapshot();
  });
});
