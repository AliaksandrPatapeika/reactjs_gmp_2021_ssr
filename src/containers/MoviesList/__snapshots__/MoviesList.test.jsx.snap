// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList renders correctly 1`] = `
<div
  className="moviesListContainer"
>
  <div
    className="movieCardContainer"
    onMouseLeave={[Function]}
  >
    <button
      className="posterButton"
      disabled={false}
      id={null}
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie card poster"
        className="movieCardPoster"
        onError={[Function]}
        src="test-file-stub"
      />
    </button>
    <button
      className="threeDotsIcon"
      disabled={false}
      id={null}
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie card menu"
        src="/img/moreButton.svg"
      />
    </button>
    <div
      className="movieTitleReleaseDateContainer"
    >
      <div
        className="movieTitle"
      >
        Pulp Fiction
      </div>
      <div
        className="movieReleaseDate"
        title="Release date: 2004-03-15"
      >
        2004
      </div>
    </div>
    <div
      className="movieGenres"
    >
      Action & Adventure
    </div>
  </div>
  <div
    className="movieCardContainer"
    onMouseLeave={[Function]}
  >
    <button
      className="posterButton"
      disabled={false}
      id={null}
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie card poster"
        className="movieCardPoster"
        onError={[Function]}
        src="test-file-stub"
      />
    </button>
    <button
      className="threeDotsIcon"
      disabled={false}
      id={null}
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie card menu"
        src="/img/moreButton.svg"
      />
    </button>
    <div
      className="movieTitleReleaseDateContainer"
    >
      <div
        className="movieTitle"
      >
        Bohemian Rhapsody
      </div>
      <div
        className="movieReleaseDate"
        title="Release date: 2003-05-23"
      >
        2003
      </div>
    </div>
    <div
      className="movieGenres"
    >
      Drama, Biography, Music
    </div>
  </div>
  <div
    className="movieCardContainer"
    onMouseLeave={[Function]}
  >
    <button
      className="posterButton"
      disabled={false}
      id={null}
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie card poster"
        className="movieCardPoster"
        onError={[Function]}
        src="test-file-stub"
      />
    </button>
    <button
      className="threeDotsIcon"
      disabled={false}
      id={null}
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie card menu"
        src="/img/moreButton.svg"
      />
    </button>
    <div
      className="movieTitleReleaseDateContainer"
    >
      <div
        className="movieTitle"
      >
        Kill Bill: Vol 2
      </div>
      <div
        className="movieReleaseDate"
        title="Release date: 1994-01-08"
      >
        1994
      </div>
    </div>
    <div
      className="movieGenres"
    >
      Oscar winning Movie
    </div>
  </div>
  <div
    className="movieCardContainer"
    onMouseLeave={[Function]}
  >
    <button
      className="posterButton"
      disabled={false}
      id={null}
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie card poster"
        className="movieCardPoster"
        onError={[Function]}
        src="test-file-stub"
      />
    </button>
    <button
      className="threeDotsIcon"
      disabled={false}
      id={null}
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie card menu"
        src="/img/moreButton.svg"
      />
    </button>
    <div
      className="movieTitleReleaseDateContainer"
    >
      <div
        className="movieTitle"
      >
        Avengers: War of Infinity
      </div>
      <div
        className="movieReleaseDate"
        title="Release date: 2004-07-21"
      >
        2004
      </div>
    </div>
    <div
      className="movieGenres"
    >
      Action & Adventure
    </div>
  </div>
  <div
    className="movieCardContainer"
    onMouseLeave={[Function]}
  >
    <button
      className="posterButton"
      disabled={false}
      id={null}
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie card poster"
        className="movieCardPoster"
        onError={[Function]}
        src="test-file-stub"
      />
    </button>
    <button
      className="threeDotsIcon"
      disabled={false}
      id={null}
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie card menu"
        src="/img/moreButton.svg"
      />
    </button>
    <div
      className="movieTitleReleaseDateContainer"
    >
      <div
        className="movieTitle"
      >
        Inception
      </div>
      <div
        className="movieReleaseDate"
        title="Release date: 2003-05-02"
      >
        2003
      </div>
    </div>
    <div
      className="movieGenres"
    >
      Action & Adventure
    </div>
  </div>
  <div
    className="movieCardContainer"
    onMouseLeave={[Function]}
  >
    <button
      className="posterButton"
      disabled={false}
      id={null}
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie card poster"
        className="movieCardPoster"
        onError={[Function]}
        src="test-file-stub"
      />
    </button>
    <button
      className="threeDotsIcon"
      disabled={false}
      id={null}
      onClick={[Function]}
      type="button"
    >
      
      <img
        alt="movie card menu"
        src="/img/moreButton.svg"
      />
    </button>
    <div
      className="movieTitleReleaseDateContainer"
    >
      <div
        className="movieTitle"
      >
        Reservoir dogs
      </div>
      <div
        className="movieReleaseDate"
        title="Release date: 1994-11-30"
      >
        1994
      </div>
    </div>
    <div
      className="movieGenres"
    >
      Oscar winning Movie
    </div>
  </div>
</div>
`;
